GANDLF/cli/huggingface_hub_handler.py

```diff
@@ -0,0 +1,52 @@
+from huggingface_hub import HfApi, snapshot_download
+from typing import List, Union
+
+
+def push_to_model_hub(
```

Comment:
So as I understand it you'll provide a CLI tool to push and pull models from the 🤗 hub. cc @wauplin wondering if it's possible to programmatically add tags?

By default, this won't track any download numbers for the various models or add specific tags which ease model discoverability (like "image-classification" etc.). For that we can open a PR similar to this one: https://github.com/huggingface/huggingface.js/pull/669, where you specify an extension which tracks downloads each time a file with the specified extension is downloaded.

Next to that, one can add code snippets when users click the "use in GaNDLF" button which you can add, as shown in this guide: https://huggingface.co/docs/hub/en/models-adding-libraries.

Reply:
@pranayasinghcsmpl @sarthakpati do you have an example of GaNDLF repo pushed on the Hub already? Agree with @NielsRogge, having it as an "officially compatible library" on the Hub would be awesome! It enables download counts + better discoverability + possible to have code snippets.

> wondering if it's possible to programmatically add tags?

Of course it is! :hugs: Check out the [ModelCard](https://huggingface.co/docs/huggingface_hub/guides/model-cards#update-metadata) guide.

Reply:
I have uploaded a test model made using gandlf. You can find it here-https://huggingface.co/pranayasingh/test_upload1/tree/main

Reply:
Thanks for the prompt responses @NielsRogge, @Wauplin and @pranayasinghcsmpl!

Reply:
So, from an organizational and requirements perspective, we want to enable anyone training models using GaNDLF to be able to upload to HF using their own org information, and not specifically under GaNDLF itself. For example, if organization `A` is training a model `X-1`, they should be able to upload this model (including the hashes of the code, and whatever else is needed to "deploy" the model) to `huggingface.co/A/X-1` (or whatever else that makes sense). 

The main reason behind this is that MLCommons does not want to be a curator of models, and only wants to enable other organizations to train their own models.

Reply:
That's exactly what the `huggingface-cli`/`huggingface_hub` are for! The suggestion from @NielsRogge is to add a `gandlf` tag to the model card metadata from generating it. This way all repos pushed to the Hub with GaNDLF will be tagged as such and searchable. It does not require to manually curate those models yourself. The download count will happen per repo, allowing you and your users to get an idea of the traction you get on the Hub.